python/cudf_polars/cudf_polars/utils/versions.py

```diff
@@ -12,11 +12,14 @@
 
 POLARS_VERSION = parse(__version__)
 
-POLARS_VERSION_GE_13 = POLARS_VERSION >= parse("1.3")
-POLARS_VERSION_GT_13 = POLARS_VERSION > parse("1.3")
-POLARS_VERSION_LT_13 = POLARS_VERSION < parse("1.3")
+POLARS_VERSION_GE_15 = POLARS_VERSION >= parse("1.5")
+POLARS_VERSION_GT_15 = POLARS_VERSION > parse("1.5")
+POLARS_VERSION_LT_15 = POLARS_VERSION < parse("1.5")
+POLARS_VERSION_GE_16 = POLARS_VERSION >= parse("1.6")
+POLARS_VERSION_GT_16 = POLARS_VERSION > parse("1.6")
+POLARS_VERSION_LT_16 = POLARS_VERSION < parse("1.6")
 
-if POLARS_VERSION < parse("1.3"):
+if POLARS_VERSION_LT_15:
```

Comment:
It would be nice if there was some way of reminding us to clean the old ones of these out whenever we bump the minimum polars version. I spent a bit trying to come up with a precommit hook that would do this for us. I ended up with basically this (run from a bash script that fails pre-commit)

```python
import toml
import re
from packaging.version import parse

file = toml.load('pyproject.toml')
dependencies = file['project']['dependencies']

polars_version = None
for dep in dependencies:
    if 'polars' in dep:
        polars_version = dep
        break


pat = r'parse\("([0-9]+\.[0-9]+(\.[0-9]+)?)"\)'
with open("./cudf_polars/utils/versions.py", 'r') as file:
    content = file.read()
    matches = re.findall(pat, content)
    versions = [match[0] for match in matches]

min_version = polars_version.split('>=')[1]

for ver in versions:
    if parse(ver) < parse(min_version):
        raise ValueError

```
We could:
- Incorporate this or something derived from it via pre-commit
- Peel this off to a separate issue if we think its important but not critically tied to this PR
- Veto this if it is neither important nor critically tied to this PR 😄

Reply:
I quite like this idea. I wouldn't tie it to this PR. Let's make a pre-commit hook with this script in a separate PR, at which point we could also generalize it so that it works for pandas as well since we have the same issues historically with filtering out these checks for old pandas versions (and in principle we could do this for other things too like pyarrow).

Reply:
#16736 16736